### PR TITLE
Add txfees query for multi denom tx fees

### PR DESCRIPTION
**Pull request by mattverse**
## Description

To be able to support multi denom tx fees in Keplr, we need a query to be able to query the needed information without extra queries or computations on pools. 

This PR includes adding a query that would include by what amount of txfees is needed when requested a denom that is not the base denom.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: proto/osmosis/txfees/v1beta1/query.proto [hunk 1]
```diff
@@ -17,6 +17,12 @@ service Query {
     option (google.api.http).get = "/osmosis/txfees/v1beta1/fee_tokens";
   }
 
+  rpc TxFeesByDenom(QueryTxFeesByDenomRequest)
```
Comment (ValarDragon) [inline proto/osmosis/txfees/v1beta1/query.proto:20]: This name doesn't sound right. Its returning the spot price

Comment (mattverse) [reply proto/osmosis/txfees/v1beta1/query.proto:20]: Hmm, yeah agreed how abt `SpotPrice`?

Comment (ValarDragon) [reply proto/osmosis/txfees/v1beta1/query.proto:20]: Oh yeah, these should be IBC denoms / that will cause issues

Comment (ValarDragon) [reply proto/osmosis/txfees/v1beta1/query.proto:20]: Perhaps `TxFeeDenomSpotPrice`? (Or I guess the tx fee part is implied by the route including `txfees`)

File: proto/osmosis/txfees/v1beta1/query.proto [hunk 2]
```diff
@@ -17,6 +17,12 @@ service Query {
...
+      returns (QueryTxFeesByDenomResponse) {
+    option (google.api.http).get =
+        "/osmosis/txfees/v1beta1/tx_fees_by_denom/{denom}";
```
Comment (alexanderbez) [inline proto/osmosis/txfees/v1beta1/query.proto:23]:
```suggestion
        "/osmosis/txfees/v1beta1/tx_fees_by_denom";
```

File: proto/osmosis/txfees/v1beta1/query.proto [hunk 3]
```diff
@@ -17,6 +17,11 @@ service Query {
     option (google.api.http).get = "/osmosis/txfees/v1beta1/fee_tokens";
   }
 
+  rpc DenomSpotPrice(QueryDenomSpotPriceRequest)
+      returns (QueryDenomSpotPriceResponse) {
+    option (google.api.http).get = "/osmosis/txfees/v1beta1/tx_fees_by_denom";
```
Comment (ValarDragon) [inline proto/osmosis/txfees/v1beta1/query.proto:22]:
```suggestion
    option (google.api.http).get = "/osmosis/txfees/v1beta1/spot_price_by_denom";
```

File: proto/osmosis/txfees/v1beta1/query.proto [hunk 4]
```diff
@@ -29,13 +34,30 @@ service Query {
...
+// price for the specified tx fee denom
+message QueryDenomSpotPriceRequest {
+  string denom = 1 [ (gogoproto.moretags) = "yaml:\"denom\"" ];
+}
+
+// QueryDenomSpotPriceRequest defines grpc response structure for querying spot
+// price for the specified tx fee denom
+message QueryDenomSpotPriceResponse {
+  uint64 poolID = 1 [ (gogoproto.moretags) = "yaml:\"pool_id\"" ];
+  string base_denom = 2 [ (gogoproto.moretags) = "yaml:\"base_denom\"" ];
```
[the remark this answers is not in the document]
Comment (ValarDragon) [reply proto/osmosis/txfees/v1beta1/query.proto:53]: base denom should always be osmo right?

[the remark this answers is not in the document]
Comment (mattverse) [reply proto/osmosis/txfees/v1beta1/query.proto]: yeah agreed!

**Review by ValarDragon — APPROVED** [review]

**Pull request merged**